lib/sharing/index.ts

```diff
@@ -0,0 +1,307 @@
+import type { CreateOptions, RequestCallback, RequestOptions } from '../types';
+
+/**
+ * Asset types supported by the Sharing API
+ */
+export enum AssetType {
+  SHEET = 'sheet',
+  REPORT = 'report',
+  SIGHT = 'sight',
+  WORKSPACE = 'workspace',
+  COLLECTION = 'collection',
+  FILE = 'file'
+}
+
+/**
+ * Access levels for sharing
+ */
+export enum AccessLevel {
+  ADMIN = 'ADMIN',
+  COMMENTER = 'COMMENTER',
+  EDITOR = 'EDITOR',
+  EDITOR_SHARE = 'EDITOR_SHARE',
+  OWNER = 'OWNER',
+  VIEWER = 'VIEWER'
+}
+
+/**
+ * Share scope
+ */
+export enum ShareScope {
+  ITEM = 'ITEM',
+  WORKSPACE = 'WORKSPACE'
+}
+
+/**
+ * Share type
+ */
+export enum ShareType {
+  GROUP = 'GROUP',
+  USER = 'USER'
+}
+
+/**
+ * Share response object
+ */
+export interface ShareResponse {
+  id: string;
+  email?: string;
+  userId?: number;
+  groupId?: number;
+  name?: string;
+  type: ShareType;
+  accessLevel: AccessLevel;
+  scope: ShareScope;
+}
+
+/**
+ * Create share request object
+ */
+export interface CreateShareRequest {
+  email?: string;
+  groupId?: number;
+  accessLevel: AccessLevel;
+  subject?: string;
+  message?: string;
+  ccMe?: boolean;
+}
+
+/**
+ * Update share request object
+ */
+export interface UpdateShareRequest {
+  accessLevel: AccessLevel;
+}
+
+/**
+ * Response for share operations
+ */
+export interface SharesResponse {
+  items: ShareResponse[];
+  lastKey?: string;
+}
+
+/**
+ * Success result with shares
+ */
+export interface SharesSuccessResult {
+  result: ShareResponse[];
+  message: string;
+  resultCode: number;
+}
+
+/**
+ * Query parameters for listing shares
+ */
+export interface ListSharesQueryParams {
+  maxItems?: number;
+  lastKey?: string;
+  include?: string;
+}
+
+/**
+ * Options for listing shares
+ */
+export interface ListSharesOptions extends RequestOptions<ListSharesQueryParams, undefined> {
+  assetType: AssetType;
+  assetId: string | number;
+  maxItems?: number;
+  lastKey?: string;
```

Comment:
since `ListSharesOptions` extends `ListSharesQueryParams` I don't think we need 
```
  maxItems?: number;
  lastKey?: string;
```

here as well. 

Later on on line 194 we deconstruct like so

```suggestion:194
--     const { assetType, assetId, maxItems, lastKey, sharingInclude } = options;
++    const { assetType, assetId, sharingInclude, queryParameters } = options;
```